### PR TITLE
[PM-5384] Add Countdown Timer to Duo Redirect

### PR DESCRIPTION
**Pull request by rr-bw**
## Type of change

- New feature development

## Objective

To add a countdown timer to the duo-redirect tab, after which the tab will automatically close.

## Screencast

In this video I run through the process 3 times:
- 1st time - Let the countdown timer expire, which closes the tab
- 2nd time - Click the close button before the countdown expires, which also closes the tab
- 3rd time - Create a handOffMessage with no countdown. In this case there will be no countdown timer and the user will have to close the tab manually.

https://github.com/bitwarden/clients/assets/102181210/598dda8f-70da-4fed-b1bb-471e06a738e7

### CODE REVIEW

File: apps/web/src/connectors/duo-redirect.ts [hunk 1]
```diff
@@ -14,23 +14,87 @@ window.addEventListener("load", () => {
...
+ *   automatically close the tab. Make sure to only add one number, as only one will be parsed.
+ *
+ * - If the <p> tag does not contain a number there will be no countdown timer and the user
+ *   will have to close the tab manually.
+ *
+ * - This implementation is intentionally simple in order to prevent the client (a) from having to add
+ *   extra "HTML elements" and (b) from having to split up the <p> text into three spans/translations...
+ *        ['This window will automatically close in', '5', 'seconds']
+ *   ...which would cause bad translations in languages that swap the order of words
+ */
```
Comment (jlf0dev) [inline apps/web/src/connectors/duo-redirect.ts:51]: fantastic documentation 👏 if you have the time it would be nice for this to use Jsdoc

**Review by jlf0dev — DISMISSED** [review]
Had a suggestion but not a blocker. Thanks for the quick work!

File: apps/web/src/connectors/duo-redirect.ts (hunk 1 above)
Comment (rr-bw) [reply apps/web/src/connectors/duo-redirect.ts:51]: @jlf0dev Great, will do. I also chatted with Jared in a call and will be making some additional changes.

**Review by JaredSnider-Bitwarden — APPROVED** [review]
Great work on this! Thanks for the screen recording of the various scenarios! 🚀

**Pull request merged**